crates/autopilot/src/database/auction.rs

```diff
@@ -73,24 +73,18 @@ impl Postgres {
         sqlx::query("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ")
             .execute(ex.deref_mut())
             .await?;
-        let orders: Vec<Order> = database::orders::solvable_orders(&mut ex, min_valid_to as i64)
-            .map(|result| match result {
-                Ok(order) => full_order_into_model_order(order),
-                Err(err) => Err(anyhow::Error::from(err)),
-            })
-            .try_collect()
-            .await?;
+        let orders: HashMap<domain::OrderUid, Order> =
+            database::orders::solvable_orders(&mut ex, min_valid_to as i64)
+                .map(|result| match result {
+                    Ok(order) => full_order_into_model_order(order)
+                        .map(|order| (domain::OrderUid(order.metadata.uid.0), order)),
+                    Err(err) => Err(anyhow::Error::from(err)),
+                })
+                .try_collect()
+                .await?;
         let latest_settlement_block =
             database::orders::latest_settlement_block(&mut ex).await? as u64;
-        let quotes = self
-            .read_quotes(
-                orders
-                    .iter()
-                    .map(|order| domain::OrderUid(order.metadata.uid.0))
-                    .collect::<Vec<_>>()
-                    .iter(),
-            )
-            .await?;
+        let quotes = self.read_quotes(orders.keys()).await?;
```

Comment:
Couldn't this `await` be parallelized with the `latest_settlement_block()` one? :thinking:

Reply:
No, since there is an opened transaction.

Reply:
I don't understand. Are you saying that in order to get the quote, we need to first read the latest settlement block?

Reply:
There is an opened tx `https://github.com/cowprotocol/services/blob/a10b4e33c7c8ed10e65d8f1348a9143ad1d80188/crates/autopilot/src/infra/persistence/mod.rs#L400-L406`. We should fetch all the data using a single DB snapshot even for quotes since there is a logic of updating them based on the onchain data: https://github.com/cowprotocol/services/blob/e7de6bfd0526e00b55fda817b33a0fde715c8a61/crates/autopilot/src/database/onchain_order_events.rs#L329-L342

Reply:
In that case, shouldn't we use the same `tx` for fetching the quote?

Reply:
Oh, right. Looks like I reverted the wrong change during one of the reworks. Thanks!